dvc/cache/local.py

```diff
@@ -24,6 +24,14 @@ def __init__(self, tree):
         super().__init__(tree)
         self.cache_dir = tree.config.get("url")
 
+        shared = tree.config.get("shared")
+        if shared:
+            self._file_mode = 0o664
+            self._dir_mode = 0o2775
+        else:
+            self._file_mode = 0o666 & ~umask
+            self._dir_mode = 0o777 & ~umask
```

Comment:
Previously we've used hardcoded modes, which was not right. Now properly using umask, same as git.